docker/bookkeeper/entrypoint.sh

```diff
@@ -74,6 +75,27 @@ create_bookie_dirs() {
   done
 }
 
+# Create a Bookie ID if this is a newly added bookkeeper pod
+# or read the Bookie ID if it already exists
+set_bookieid() {
+  IFS=',' read -ra directories <<< $1
+  BK_ID_DIR="${directories[0]}"
+  BK_ID_FILE="${BK_ID_DIR}/id"
+  if [ `find ${BK_ID_FILE} | wc -l` -gt 0 ]; then
+    echo "Found BookieID"
+    BK_bookieId=`cat ${BK_ID_FILE}`
+  else
+    echo "BookieID not found"
+    if [[ $BK_setNewBookieID == "true" ]];then
+      BK_bookieId="`hostname -s`-`date "+%Y%m%d-%H%M%S"`"
+    else
+      HOST="$(echo -e `hostname -A` | sed -e 's/[[:space:]]*$//')"
```

Comment:
If the Operator permitted to change the option `useHostNameAsBookieID`, then this would be problematic, as you wouldn't know if it is needed to set the hostname or the IP. I think that the best way to set the previous ID of the Bookie is just to get it from the Cookie, if there is one. I know that we have not seen any command like "get me the ID from the Cookie for this Bookie", but that would be the most reliable approach I guess.

Reply:
@RaulGracia [this](https://github.com/pravega/bookkeeper-operator/pull/166) PR also includes the change to disallow users from modifying the value of the option `useHostNameAsBookieID`. This would be the only way to set the previous ID since getting it from the cookie would not be possible till the connection with zookeeper cluster has been established, whereas the environment variable `bookieId` would need to be set and exposed before doing that.